templates/powershell/modules/app-config/app-config.psm1

```diff
@@ -439,9 +538,24 @@ function Import-AppConfigValues {
 
 	process {
 		Write-Debug "${functionName}:process:Start"
-
 		[AppConfigEntry[]]$existingItems = Get-AppConfigValues -ConfigStore $ConfigStore -Label $Label
-		[AppConfigEntry[]]$desiredItems = Get-AppConfigValuesFromFile -Path $importFile.FullName
+		if ($importFile.FullName.EndsWith(".json")) {
+			[AppConfigEntry[]]$desiredItems = Get-AppConfigValuesFromFile -Path $importFile.FullName
+		}
+		elseif ($importFile.FullName.EndsWith(".yaml")) {
+			[AppConfigEntry[]]$desiredItems = Get-AppConfigValuesFromYamlFile -Path $importFile.FullName -DefaultLabel $Label -KeyVault $KeyVaultName 
+		}
+		else {
+			throw [System.IO.InvalidDataException]::new($importFile.FullName)
+		}
+
+		#Validate if each record has a label matching the service
+		foreach ($item in $desiredItems) {
```

Comment:
Can you change the loop to | foreach-object
Same as the next function.

Reply:
Updated